NETCORE/src/Shared/Extensions/ApplicationInsightsServiceOptions.cs

```diff
@@ -97,7 +69,7 @@ public ApplicationInsightsServiceOptions()
         /// <summary>
         /// Gets or sets the application version reported with telemetries.
         /// </summary>
-        public string ApplicationVersion { get; set; }
+        public string ApplicationVersion { get; set; } = Assembly.GetEntryAssembly()?.GetName().Version.ToString();
```

Comment:
Does this mean `Assembly.GetEntryAssembly()?.GetName().Version.ToString()` gets executed everytime one accesses ApplicationVersion? Previously this was executed once inside ctor only.

Reply:
Nope, this just defines the default values for these properties. This is only executed when a new instance is created. It will be the exact same behavior as the former default constructor.

Reply:
> Auto-property initializers let you declare the initial value for an auto-property as part of the property declaration.
> https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-6#auto-property-initializers